### ISSUE TITLE
Better representation of diffs for map when values are different though key is same.

### ISSUE THREAD

**Issue by cocuh** [google/assertor#38]
Derived from #30.  

Current formatting logic for different values can be ugly in some case; specifically 1) when formatted string length of `key` is quite long, 2) the formatted string length of `values` are quite long, 3) there are many diffs.

**Current:** 
```
key {:?} was mapped to an unexpected value
expected value {:?}, but found {:?}
```

**Improvement idea from [this comment](https://github.com/google/assertor/issues/30#issuecomment-1629704493)**
```
keys were mapped to an unexpected values: [
  "key1": {
    "expected" : "val1",
    "actual"      : "val2" 
  }
  ...
]
```

**Comment by kelebra** [comment]
This should probably be closed after PR #56  :)

**Comment by cocuh** [comment]
Yes, thanks!
